VeronicaM/residence
# feat: HeatMap

# HeatMap

## What is this change ?

- Add heatmap component using `google-maps-react` npm package
- Protect API key from being stored in repository by passing it as env variable 
- Protect API key from being used unreasonably by restricting it by referrer to currently deployed to app https://residence-veram.netlify.app/ and to http://localhost:3001 for testing purposes. 

## How to test
- Fill in your GOOGLE_MAPS_API_KEY value in .env.local
- run the app with `yarn start`
- Check that the heat map renders correctly with the hardcoded residence points from db.json
- Check that the heat map re-renders after a new residence entry is added to the db. 
- run `yarn test`, all tests should pass
- run `yarn lint`, there should be no lint error
